### PR TITLE
helm: use tpl for ingress hosts

### PR DESCRIPTION
**Pull request by christianjedroCDT**
**What this PR does / why we need it**: Use tpl for ingress hosts. Needed for values passed from globals in umbrella charts. We´re already using this in [Grafana Helm Chart](https://github.com/grafana/helm-charts/blob/d8a64071970b5928cf990dce480925ff6b8edb56/charts/grafana/templates/ingress.yaml#L37). 

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [ ] Tests updated
- [x] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [x] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)

### CODE REVIEW

File: docs/sources/setup/install/helm/reference.md [hunk 1]
```diff
@@ -1094,7 +1094,7 @@ false
 		<tr>
 			<td>gateway.ingress.hosts</td>
 			<td>list</td>
-			<td>Hosts configuration for the gateway ingress</td>
+			<td>Hosts configuration for the gateway ingress. Host property will be passed through `tpl`.</td>
```
[the remark this answers is not in the document]
Comment (JStickler) [reply docs/sources/setup/install/helm/reference.md]: I actually think that makes it less clear.   In the first draft "will be passed" made it sound like using a Helm template was a requirement.  The second draft removed the definition for the key. I think what you want is something like this:

<td>Hosts configuration for the gateway ingress. Host property can be configured in the chart or passed through a Helm template.</td>

[the remark this answers is not in the document]
Comment (christianjedroCDT) [reply docs/sources/setup/install/helm/reference.md]: I´ve actually found an additional ingress which needs the `tpl` function. Also i´ve moved to an explanation of the 
`tpl` function which was already used in the same file for the same usecase: https://github.com/grafana/loki/blob/main/production/helm/loki/values.yaml#L1337

Please let me know if that´s fine, thanks!

[the remark this answers is not in the document]
Comment (JStickler) [reply docs/sources/setup/install/helm/reference.md]: That reads much clearer to me.  👍  But I'll leave it up to one of the engineers to approve the PR for merging (I'm the technical writer).

**Comment by christianjedroCDT** [comment]
@ashwanthgoli could you please have a look into this?

**Comment by christianjedroCDT** [comment]
> @christianjedroCDT Thanks for this contribution. Can you fix the merge conflicts and any CI issues? Most likely you'll need to pick a later version.

done :)

**Review by MichelHollands — APPROVED** [review]

**Pull request merged**